src/grammar.rs

```diff
@@ -165,6 +169,56 @@ impl<'gram> fmt::Display for ParseTree<'gram> {
     }
 }
 
+pub struct MermaidParseTree<'a> {
```

Comment:
because a single type like `ParseTree` cannot `impl std::fmt::Display` twice, this wrapper type is how a parse tree can be opted into Mermaid display.